### PR TITLE
Poetry Installation Update and Dependencies Update

### PR DESCRIPTION
**Pull request by td928**
#350 which was flagged earlier in  #347  and #326 

This installation update is fairly simple and we have done very similar works on [gru qaqc recently](https://github.com/NYCPlanning/db-gru-qaqc/pull/211).  On other other hand, one unexpected complexity is after poetry installation is updated and the existing `poetry.lock` no longer works. This leads me down in a deep rabbit hole on learning about poetry maintenance and [Python dependencies shenanigans](https://dustingram.com/articles/2018/03/05/why-pypi-doesnt-know-dependencies/). I will share some of my most important learnings here. 

## `poetry install/lock -vv`
After reading [a bunch](https://github.com/python-poetry/poetry/issues/2094#issuecomment-609648194) on the poetry slow resolve dependency issues, I finally stumbled across this useful command with `poetry install -vv`. What it does is showing the verbose details for poetry resolving dependencies and some comments from post linked also highlighted that _poetry is essentially going through each possible version and figuring out the dependency graph for each version_. One can imagine this becomes **very expensive** if the lower bound version to be very far from the latest. So the trial and error way for me to solve this is 
1. run this command with verbose option and then see which dependencies it got hang on. Terminates the process by aborting (control + c) if needed
2. raise the lower bounds for that package close to latest version 
3. rerun `poetry install` to see if it still hangs. After a few tries with this, the dependencies resolving time dropped to around 10 seconds or so. 

## gdal
updated to latest `3.6.1`

### jinja2 
unrelated but there is also [a issue](https://stackoverflow.com/a/73726848) after everything else is reinstalled so it is also updated. 

## Notes for testing
I run some datasets to test locally whether the functionalities for data library is working properly. For example I was running `library archive -n dcp_nta2020 -v 22c`

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
 RUN apt update && apt install -y python3-pip python3-distutils git
 
 RUN adduser --disabled-password --gecos '' vscode
 
 USER vscode
 
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | POETRY_VERSION=1.1.6 python3 -
-
-ENV PATH="~/.poetry/bin:$PATH"
-
```
Comment (mbh329) [inline .devcontainer/Dockerfile:12]: How come we don't have to set the env. variable paths? Is the default that poetry knows where  to install itself? I am still confused about this

Comment (mbh329) [reply .devcontainer/Dockerfile:12]: Or is it just sent in .json now?

**Comment by mbh329** [comment]
The test of the dcp_nta2020 data worked on my local machine but going to do some more testing so that the new packages work as expected

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (td928) [reply .devcontainer/Dockerfile:12]: yeah I think what is in the .json is sufficient for setting up the PATH variable

Comment (mbh329) [reply .devcontainer/Dockerfile:12]: I think I just want to get on the same page about where to specify the PATH and then I can approve this. Tested a bunch of different datasets and the outputs look good

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+FROM osgeo/gdal:ubuntu-small-3.6.1
 
 COPY . /library/
 
 WORKDIR /library/
 
 RUN apt update && apt install -y python3-pip python3-distutils
 
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | POETRY_VERSION=1.1.6 python3 -
+RUNcurl -sSL https://install.python-poetry.org | python3 -
```
Comment (mbh329) [inline Dockerfile:9]: space between `RUN` and `CURL`

Comment (td928) [reply Dockerfile]: fixed this in the new commit

**Review by mbh329 — APPROVED** [review]
LGTM

**Pull request merged**